### PR TITLE
[WIP] Add a read only URL field when the strategy is receive

### PR DESCRIPTION
**Pull request by rohithasrk**
References #9.This PR has a lot more work to do. As of now, I only followed the instructions given in #9 comments. There is a problem with my understanding of read only fields in django admin. Here are my doubts:
  - The read only field doesn't appear in the admin site as of now. Does a HTML element automatically get created with `id = id_<parameter>`? This is what I was able to make out checking the code.
  - And regarding the js file similar to uuid.js in django_netjsonconfig, I made some changes accordingly but I'm not so clear about it.

@nemesisdesign Please review and study material suggestions are requested.

### CODE REVIEW

**Comment by nemesisdesign** [comment]
@rohithasrk need to add explicit `fields` declaration when adding extra fields.

Also don't forget to load the static files by specifying the `Media` class. I've added a bit of CSS to make the input look similar to what we have in OpenWISP2 (uuid).

Please rename `post_url` to `receive_url` and use dashes instead of underscores for the name of static files (`receive-url.js`).

**Comment by nemesisdesign** [comment]
Now use django.urls.reverse to get the URL and append the key on its querystring

File: django_netjsongraph/admin.py [hunk 1]
```diff
@@ -9,9 +11,21 @@
...
+    fields = ['label', 'parser', 'strategy', 'url', 'key',
+              'expiration_time', 'receive_url', 'published', 'protocol',
+              'version', 'revision', 'metric', 'created']
+
+    class Media:
+        css = {'all': [static('netjsongraph/admin.css')]}
+        js = [static('netjsongraph/receive-url.js')]
+
+    def receive_url(self, obj):
+        return reverse('receive_topology/', kwargs={'pk': obj.key})
```
Comment (nemesisdesign) [inline django_netjsongraph/admin.py]: This is wrong, please use obj.pk and append obj.key as its querystring

File: django_netjsongraph/admin.py [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+
+    receive_url.short_description = "RECEIVE_URL"
```
Comment (nemesisdesign) [inline django_netjsongraph/admin.py]: please use a human readable string translated with gettext, eg: `_('receive url')`

File: django_netjsongraph/admin.py [hunk 3]
```diff
@@ -1,17 +1,36 @@
...
+              'version', 'revision', 'metric', 'created']
+
+    class Media:
+        css = {'all': [static('netjsongraph/admin.css')]}
+        js = [static('netjsongraph/receive-url.js')]
+
+    def receive_url(self, obj):
+        return reverse('receive_topology/', kwargs={'pk': obj.pk}) + obj.key
+
+    receive_url.short_description = "RECEIVE_URL"
```
Comment (nemesisdesign) [inline django_netjsongraph/admin.py]: please use a human readable string translated with gettext, eg: `_('receive url')`

File: django_netjsongraph/admin.py [hunk 4]
```diff
@@ -1,17 +1,36 @@
...
+    fields = ['label', 'parser', 'strategy', 'url', 'key',
+              'expiration_time', 'receive_url', 'published', 'protocol',
+              'version', 'revision', 'metric', 'created']
+
+    class Media:
+        css = {'all': [static('netjsongraph/admin.css')]}
+        js = [static('netjsongraph/receive-url.js')]
+
+    def receive_url(self, obj):
+        return reverse('receive_topology/', kwargs={'pk': obj.pk}) + obj.key
```
Comment (nemesisdesign) [inline django_netjsongraph/admin.py]: wrong again, use:

```python
return '{0}?key={1}'.format(reverse('receive_topology/', kwargs={'pk': obj.pk}), obj.key)
```

File: django_netjsongraph/admin.py (hunk 3 above)
Comment (rohithasrk) [inline django_netjsongraph/admin.py]: I see. Thanks

**Pull request merged**